src/Microsoft.Build.Sql/NugetClient.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Versioning;
+
+namespace Microsoft.Build.Sql;
+
+public static class NugetClient
+{
+    private const string CacheFileName = "VersionCache.json";
+    private const int CacheFileExpirationInDays = 7;
+    private const int HttpClientTimeoutInSeconds = 5;
+
+    /// <summary>
+    /// Factory method to create an HttpClient instance, defaults to a new instance with a 5-second timeout.
+    /// </summary>
+    public static Func<HttpClient> HttpClientFactory { get; set; } = () => new HttpClient() { Timeout = TimeSpan.FromSeconds(HttpClientTimeoutInSeconds) };
+
+    /// <summary>
+    /// Gets the latest version of given package. If there is a cached version, it will be used if not expired.
+    /// If the cache is expired or does not exist, it will fetch the latest version from NuGet.org.
+    /// </summary>
+    public static async Task<NuGetVersion> GetLatestPackageVersion(string packageName, bool prerelease, CancellationToken cancellationToken = default)
+    {
+        // Check if we have a cached version
+        NuGetVersionsData versionsData = await GetCachedVersionDataAsync(packageName, cancellationToken);
+
+        // If not, fetch the latest version from NuGet.org
+        if (versionsData == null)
+        {
+            // API: https://learn.microsoft.com/nuget/api/package-base-address-resource#enumerate-package-versions
+            string url = $"https://api.nuget.org/v3-flatcontainer/{packageName.ToLowerInvariant()}/index.json";
+
+            using HttpClient client = HttpClientFactory();
+            using HttpResponseMessage response = await client.GetAsync(url, cancellationToken);
```

Comment:
can set timeout?

Reply:
It's set to 5 seconds in the factory on line 24